process/sync/storageBootstrap/baseStorageBootstrapper_test.go

```diff
@@ -401,3 +401,74 @@ func TestBaseStorageBootstrapper_GetBlockBodyShouldWork(t *testing.T) {
 	assert.Nil(t, err)
 	assert.Equal(t, expectedBody, body)
 }
+
+func TestBaseStorageBootstrapper_setCurrentBlockInfoV3(t *testing.T) {
+	t.Parallel()
+
+	t.Run("in case of nil LastExecutionResult should fail", func(t *testing.T) {
+		t.Parallel()
+
+		baseArgs := createMockShardStorageBootstrapperArgs()
+		args := ArgsShardStorageBootstrapper{
+			ArgsBaseStorageBootstrapper: baseArgs,
+		}
+
+		ssb, _ := NewShardStorageBootstrapper(args)
+		err := ssb.setCurrentBlockInfoV3(&block.HeaderV3{
+			LastExecutionResult: nil,
+		}, nil)
+
+		require.Equal(t, process.ErrNilLastExecutionResultHandler, err)
+	})
+
+	t.Run("if getting the header fails, the error should be propagated", func(t *testing.T) {
+		t.Parallel()
+
+		errExpected := errors.New("expected error")
+		baseArgs := createMockShardStorageBootstrapperArgs()
+		baseArgs.Marshalizer = &testscommon.MarshallerStub{
+			UnmarshalCalled: func(obj interface{}, buff []byte) error {
+				return errExpected
+			},
+		}
+		args := ArgsShardStorageBootstrapper{
+			ArgsBaseStorageBootstrapper: baseArgs,
+		}
+
+		ssb, _ := NewShardStorageBootstrapper(args)
+		err := ssb.setCurrentBlockInfoV3(&block.HeaderV3{
+			LastExecutionResult: &block.ExecutionResultInfo{
+				ExecutionResult: &block.BaseExecutionResult{
+					HeaderHash: []byte("hashExecResult"),
+				},
+			},
+		}, []byte("hash"))
+
+		require.Equal(t, process.ErrUnmarshalWithoutSuccess, err)
+	})
+
+	t.Run("should work", func(t *testing.T) {
+		t.Parallel()
+
+		baseArgs := createMockShardStorageBootstrapperArgs()
+		baseArgs.Marshalizer = &testscommon.MarshallerStub{
+			UnmarshalCalled: func(obj interface{}, buff []byte) error {
+				return nil
+			},
+		}
+		args := ArgsShardStorageBootstrapper{
+			ArgsBaseStorageBootstrapper: baseArgs,
```

Comment:
```suggestion
		baseArgs := createMockShardStorageBootstrapperArgs()

		args := ArgsShardStorageBootstrapper{
			ArgsBaseStorageBootstrapper: baseArgs,
```
marshaller code here is not needed on happy flow

Reply:
the Unmarshal method has to be defined to avoid returning error